packages/sdk-middleware-auth/src/build-requests.js

```diff
@@ -43,8 +46,48 @@ export function buildRequestForClientCredentialsFlow (
   return { basicAuth, url, body }
 }
 
-export function buildRequestForPasswordFlow () {
-  // TODO
+export function buildRequestForPasswordFlow (
+  options: PasswordAuthMiddlewareOptions,
+): BuiltRequestParams {
+  if (!options)
+    throw new Error('Missing required options')
+
+  if (!options.host)
+    throw new Error('Missing required option (host)')
+
+  if (!options.projectKey)
+    throw new Error('Missing required option (projectKey)')
+
+  if (!options.credentials)
+    throw new Error('Missing required option (credentials)')
+
+  const {
+    clientId,
+    clientSecret,
+    user,
+  } = options.credentials
+  const pKey = options.projectKey
+  if (!(clientId && clientSecret && user))
+    throw new Error(
+      'Missing required credentials (clientId, clientSecret, user)',
+    )
+  const { username, password } = user
+  if (!(username && password))
+    throw new Error('Missing required user credentials (username, password)')
+
+  const scope = (options.scopes || []).join(' ')
+  const scopeStr = scope ? `&scope=${scope}` : ''
+
+
+  const basicAuth = new Buffer(`${clientId}:${clientSecret}`).toString('base64')
+  // This is mostly useful for internal testing purposes to be able to check
+  // other oauth endpoints.
+  const oauthUri = options.oauthUri || `/oauth/${pKey}/customers/token`
+  const url = options.host.replace(/\/$/, '') + oauthUri
+  // eslint-disable-next-line max-len
+  const body = `grant_type=password&username=${username}&password=${password}${scopeStr}`
```

Comment:
Since this is in the main code (that is being used in the browser also), I don't think it make sense to add a **new** module to the build just to trim new line.

Not a valid trade off from my point of view.